### PR TITLE
RTable getField improvment

### PR DESCRIPTION
**Pull request by Mathewlenning**

### CODE REVIEW

File: extensions/libraries/redcore/table/nested.php [hunk 1]
```diff
@@ -787,4 +787,115 @@ public function hasPrimaryKey()
...
+	 * @param   \JDate  $now        the current time
+	 *
+	 * @return array
+	 */
+	private function updateSchema($assetName, \JDate $now)
+	{
+		$dbo = $this->getDbo();
+		$query = $dbo->getQuery(true);
+
+		$query->update('#__logical_schemas');
```
Comment (shandak) [inline extensions/libraries/redcore/table/nested.php]: I didn`t check how it works, but seems like it is wront table name, it should be `#__redcore_schemas`

File: extensions/libraries/redcore/table/table.php [hunk 2]
```diff
@@ -1032,4 +1032,115 @@ public static function updateAuditFields(&$tableInstance)
...
+	 * @param   \JDate  $now        the current time
+	 *
+	 * @return array
+	 */
+	private function updateSchema($assetName, \JDate $now)
+	{
+		$dbo = $this->getDbo();
+		$query = $dbo->getQuery(true);
+
+		$query->update('#__logical_schemas');
```
Comment (shandak) [inline extensions/libraries/redcore/table/table.php]: Same problem

File: extensions/libraries/redcore/table/nested.php (hunk 1 above)
Comment (Mathewlenning) [inline extensions/libraries/redcore/table/nested.php]: Nice catch :smile: This is one of the reasons I hate duplication, are we planning on using traits in redcore any time soon? that would help keep these improvements DRY

Comment (shandak) [inline extensions/libraries/redcore/table/nested.php]: I think you can use trait if you want, us I know we don't have restrictions about it.
I used it successfull in redSHOPB for tables https://github.com/redCOMPONENT-COM/redSHOPB2B/blob/develop/extensions/libraries/redshopb/table/webservices.php#L19

**Comment by Mathewlenning** [comment]
@Kixo @phproberto I just noticed this PR is still pending, anything I need to change to get this merged?

File: extensions/components/com_redcore/admin/sql/install/mysql/install.sql [hunk 3]
```diff
@@ -297,3 +297,14 @@ CREATE TABLE IF NOT EXISTS `#__redcore_translation_columns` (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
 SET FOREIGN_KEY_CHECKS = 1;
+
+-- -----------------------------------------------------
+-- Table `#__redcore_schemas`
+-- -----------------------------------------------------
+
+CREATE TABLE IF NOT EXISTS `#__redcore_schemas` (
+  `asset_id` varchar(255) COLLATE utf8_unicode_ci NOT NULL,
```
Comment (Kixo) [inline extensions/components/com_redcore/admin/sql/install/mysql/install.sql]: why are you forcing collation? It should be db default

**Comment by Kixo** [comment]
can you please explain where is this needed exactly?
This is caching of current table fields into json for the 1 day duration...
why is this more helpful/faster than fetching it with getTableColumns method every time since it is 1 query either way...

**Comment by Mathewlenning** [comment]
@Kixo this is a big performance booster. Especially if you're using multiple tables per cycle. ```SHOW FULL COLUMNS``` avg between 4.5ms and 6ms per query while select is usually between 0.20ms to 1.5ms

File: extensions/components/com_redcore/admin/sql/install/mysql/install.sql (hunk 3 above)
Comment (Kixo) [inline extensions/components/com_redcore/admin/sql/install/mysql/install.sql]: can you please not force collation :)

**Comment by Mathewlenning** [comment]
oops forgot to push the changes! fixed =^D

File: extensions/redcore.xml [hunk 4]
```diff
@@ -7,7 +7,7 @@
     <authorUrl>www.redcomponent.com</authorUrl>
     <copyright>Copyright (C) 2008 - 2016 redCOMPONENT.com. All rights reserved.</copyright>
     <license>GNU General Public License version 2 or later, see LICENSE.</license>
-    <version>1.8.9</version>
+    <version>1.8.10</version>
```
Comment (Kixo) [inline extensions/redcore.xml]: can you please set version to 1.9.0

Comment (Mathewlenning) [inline extensions/redcore.xml]: 👍

File: extensions/redcore.xml [hunk 5]
```diff
@@ -57,7 +57,7 @@
         <menu view="dashboard">COM_REDCORE</menu>
         <submenu>
             <menu link="option=com_redcore&amp;view=dashboard" view="dashboard">COM_REDCORE_DASHBOARD</menu>
-            <menu link="option=com_redcore&amp;view=config&amp;layout=edit&amp;component=com_redcore" view="configs">COM_REDCORE_CONFIGURATION</menu>
+            <menu link="option=com_redcore&amp;view=configs" view="configs">COM_REDCORE_CONFIGURATION</menu>
```
Comment (Kixo) [inline extensions/redcore.xml]: actually this link should go directly to Config, like it was, why are you changing it? :)

Comment (Mathewlenning) [inline extensions/redcore.xml]: Actually I don't remember changing that, might be a botched merge

**Pull request merged**